src/main/java/net/dries007/tfc/common/entities/Fauna.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.entities;
+
+import java.util.Objects;
+
+import com.google.gson.JsonObject;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.SpawnPlacements;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.Fluids;
+
+import net.minecraftforge.registries.ForgeRegistries;
+
+import com.mojang.serialization.JsonOps;
+import net.dries007.tfc.world.chunkdata.ChunkData;
+import net.dries007.tfc.world.decorator.ClimateConfig;
+
+/**
+ * A data driven way to make spawning conditions for animals player configurable.
+ * //todo: In 1.18, spawn predicates are set to change. We will have to adapt or throw this out.
+ */
+public class Fauna
+{
+    private final ResourceLocation id;
+    private final EntityType<Mob> entity;
+    private final Fluid fluid;
+    private final int chance;
+    private final int distanceBelowSeaLevel;
+    private final ClimateConfig climateConfig;
+    private final boolean solidGround;
+
+    @SuppressWarnings("unchecked")
+    public Fauna(ResourceLocation id, JsonObject json)
+    {
+        this.id = id;
+        entity = (EntityType<Mob>) ForgeRegistries.ENTITIES.getValue(new ResourceLocation(GsonHelper.getAsString(json, "entity")));
```

Comment:
```suggestion
        entity = (EntityType<Mob>) JsonHelpers.getRegistryEntry(json, "entity", ForgeRegistries.ENTITIES);
```

Similar for below. In addition, as a tip, you can call all vanilla `GsonHelper` methods through `JsonHelpers` as well, i.e. `JsonHelpers.getAsInt()` works fine.